File: Algorithms/leetcode_java/problem136/Solution1.java
package leetcode_java.problem136;

import java.util.Set;
import java.util.HashSet;

class Solution1 {
    public int singleNumber(int[] nums) {
        Set<Integer> ans = new HashSet<>();
        
        for(int num : nums){
            if(ans.contains(num)){
                ans.remove(num);
            }else{
                ans.add(num);
            }
        }
        return ans.iterator().next();
    }
}

File: Algorithms/leetcode_py/problem136/solution1.py
class Solution:
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = set()
        for num in nums:
            if num in ans:
                ans.remove(num)
            else:
                ans.add(num)
        return ans.pop()